sdk/scheduler/src/main/java/com/mesosphere/sdk/state/StateStore.java

```diff
@@ -433,6 +440,92 @@ public void clearProperty(final String key) throws StateStoreException {
+     */
+    public void storeGoalOverrideStatus(String taskName, GoalStateOverride.Status status)
+            throws StateStoreException {
+        try {
+            Map<String, byte[]> values = new TreeMap<>();
+
+            if (GoalStateOverride.Status.INACTIVE.equals(status)) {
+                // Mark inactive state by clearing any override bits.
+                values.put(getGoalOverridePath(taskName), null);
+                values.put(getGoalOverrideStatusPath(taskName), null);
+            } else {
+                values.put(getGoalOverridePath(taskName),
+                        status.target.getSerializedName().getBytes(StandardCharsets.UTF_8));
+                values.put(getGoalOverrideStatusPath(taskName),
+                        status.progress.getSerializedName().getBytes(StandardCharsets.UTF_8));
+            }
+            persister.setMany(values);
+        } catch (PersisterException e) {
+            throw new StateStoreException(e);
+        }
+    }
+
+    /**
+     * Retrieves the goal state override status of a particular task. A lack of override will result in a
+     * {@link GoalOverrideStatus} with {@code override=NONE} and {@code state=NONE}.
+     *
+     * @throws StateStoreException in the event of a storage error
+     */
+    public GoalStateOverride.Status fetchGoalOverrideStatus(String taskName) throws StateStoreException {
+        try {
+            String goalOverridePath = getGoalOverridePath(taskName);
+            String goalOverrideStatusPath = getGoalOverrideStatusPath(taskName);
+            Map<String, byte[]> values = persister.getMany(Arrays.asList(goalOverridePath, goalOverrideStatusPath));
+            byte[] nameBytes = values.get(goalOverridePath);
+            byte[] statusBytes = values.get(goalOverrideStatusPath);
+            if (nameBytes == null && statusBytes == null) {
+                // Cleared override bits => Inactive state
+                return GoalStateOverride.Status.INACTIVE;
+            } else if (nameBytes == null || statusBytes == null) {
+                throw new StateStoreException(Reason.LOGIC_ERROR,
+                        String.format("Task is missing override name or override status. " +
+                                "Expected either both or neither: %s", values));
+            }
+            return parseOverrideName(nameBytes).newStatus(parseOverrideProgress(statusBytes));
+        } catch (PersisterException e) {
+            throw new StateStoreException(e);
+        }
+    }
+
+    private static GoalStateOverride parseOverrideName(byte[] nameBytes) throws StateStoreException {
+        String name = new String(nameBytes, StandardCharsets.UTF_8);
+        for (GoalStateOverride override : GoalStateOverride.values()) {
+            if (override.getSerializedName().equals(name)) {
+                return override;
+            }
+        }
+        throw new StateStoreException(Reason.SERIALIZATION_ERROR,
```

Comment:
Again, I'm not sure we want to throw and fail in these scenarios.  Willing to be convinced otherwise, but maybe logging errors is enough with fall back to `INACTIVE` for safety sake.  How would we get out of this situation?  `podtool`?

Reply:
Very good point. This should fall back to inactive when the name isn't known, to avoid causing undue problems during a downgrade. Switched to logging a warning and returning the inactive state.